### ISSUE TITLE
[Review] Assignment 3

### ISSUE THREAD

**Issue by thanhtan1105** [vulong264/twitterclone#1]
:+1: nice work. The point of this homework was to explore a simple example of a full MVC application with a RESTful API. 

## Grading Summary
* **Functionality**: Excellent
* **Code Style**: Good
* **Visual Design**: Good

## Detailed Notes

* Nice work encapsulating the properties into the private class extension for better encapsulation.
* Nice work using private scope for your properties and your functions for better encapsulation.
* You should always specify Auto Layout constraints in your nibs.
* Nice work specifying the Auto Layout constraints.
* Nice work handling the retweet label by dynamically modifying the constraint. This is a situation where its not possible to specify the behavior that you want solely using Interface Builder.
* Nice work setting up the controls within the Tweet cells. It can sometimes be hard to have a clean design for embedded controls because it usually involves cumbersome plumbing. This is an instance where I think having the model handle the network action helps greatly simplify the implementation.
* Nice work handling the post tweet insertion into the feed. In real MVC apps, the more polished experiences have to do a certain amount of local object management to make a fully seamless experience.
* Basically, you should `deselect` the cell when back to previous screen
<img width="319" alt="screenshot 2017-03-21 16 54 48" src="https://cloud.githubusercontent.com/assets/12711672/24141788/2866eace-0e57-11e7-9b03-94b56bf1a727.png">

* For tweet view, instead of using `UITextField`, you should use `UITextView` so that user can input in multiple line.
<img width="317" alt="screenshot 2017-03-21 16 54 03" src="https://cloud.githubusercontent.com/assets/12711672/24143774/5ea705c8-0e5d-11e7-861f-a87c8db72e2b.png">

* Following Swift 3 convention, it should be in the same line
<img width="597" alt="screenshot 2017-03-21 16 53 06" src="https://cloud.githubusercontent.com/assets/12711672/24143827/8d2d3eee-0e5d-11e7-92ec-163622bf09f4.png">
